llvm/test/DebugInfo/AArch64/instr-ref-target-hooks-sp-clobber.mir

```diff
@@ -0,0 +1,245 @@
+    liveins: $w2, $x0, $x1, $x3, $x4, $x5, $x6, $x7, $x27, $x28, $x25, $x26, $x23, $x24, $x21, $x22, $x19, $x20, $lr
+  
+    $sp = frame-setup SUBXri $sp, 112, 0
+    frame-setup STPXi killed $x28, killed $x27, $sp, 2 :: (store (s64) into %stack.12), (store (s64) into %stack.11)
+    frame-setup STPXi killed $x26, killed $x25, $sp, 4 :: (store (s64) into %stack.10), (store (s64) into %stack.9)
+    frame-setup STPXi killed $x24, killed $x23, $sp, 6 :: (store (s64) into %stack.8), (store (s64) into %stack.7)
+    frame-setup STPXi killed $x22, killed $x21, $sp, 8 :: (store (s64) into %stack.6), (store (s64) into %stack.5)
+    frame-setup STPXi killed $x20, killed $x19, $sp, 10 :: (store (s64) into %stack.4), (store (s64) into %stack.3)
+    frame-setup STPXi $fp, killed $lr, $sp, 12 :: (store (s64) into %stack.2), (store (s64) into %stack.1)
+    frame-setup CFI_INSTRUCTION def_cfa_offset 112
+    frame-setup CFI_INSTRUCTION offset $w30, -8
+    frame-setup CFI_INSTRUCTION offset $w29, -16
+    frame-setup CFI_INSTRUCTION offset $w19, -24
+    frame-setup CFI_INSTRUCTION offset $w20, -32
+    frame-setup CFI_INSTRUCTION offset $w21, -40
+    frame-setup CFI_INSTRUCTION offset $w22, -48
+    frame-setup CFI_INSTRUCTION offset $w23, -56
+    frame-setup CFI_INSTRUCTION offset $w24, -64
+    frame-setup CFI_INSTRUCTION offset $w25, -72
+    frame-setup CFI_INSTRUCTION offset $w26, -80
+    frame-setup CFI_INSTRUCTION offset $w27, -88
+    frame-setup CFI_INSTRUCTION offset $w28, -96
+    DBG_PHI $x1, 1
+    $x19 = ORRXrs $xzr, killed $x7, 0
+    $x20 = ORRXrs $xzr, killed $x6, 0
+    $x21 = ORRXrs $xzr, killed $x5, 0
+    $x22 = ORRXrs $xzr, killed $x4, 0
+    $x23 = ORRXrs $xzr, killed $x3, 0
+    $w25 = ORRWrs $wzr, killed $w2, 0
+    $x26 = ORRXrs $xzr, killed $x0, 0
+    renamable $w27 = MOVZWi 1, 0
+    STRXui killed $x1, $sp, 1 :: (store (s64) into %stack.0)
+    TBNZW renamable $w25, 0, %bb.2
+  
+  bb.3.if.end16:
+    successors: %bb.2(0x04000000), %bb.3(0x7c000000)
+    liveins: $w25, $w27, $x19, $x20, $x21, $x22, $x23, $x26
+  
+    $x28 = ORRXrs $xzr, $xzr, 0
+    renamable $x24 = LDRXui killed renamable $x28, 0 :: (load (s64) from `ptr null`)
+    BL @_ZdlPvm, csr_darwin_aarch64_aapcs, implicit-def dead $lr, implicit $sp, implicit-def $sp
+    STRXui renamable $x21, renamable $x20, 0 :: (store (s64) into %ir.0)
+    STRXui killed renamable $x24, renamable $x19, 0 :: (store (s64) into %ir.1)
+    STRXui $xzr, renamable $x26, 0 :: (store (s64) into %ir.__first)
+    STRWui $wzr, renamable $x23, 0 :: (store (s32) into %ir.__first.addr.0, align 8)
+    STRWui renamable $w27, renamable $x22, 0 :: (store (s32) into %ir.Label3.i.i.i241)
+    TBZW renamable $w25, 0, %bb.3
+  
+  bb.2.if.then13:
+    liveins: $x26
+  
+    DBG_INSTR_REF !10, !DIExpression(DW_OP_LLVM_arg, 0), dbg-instr-ref(1, 0), debug-location !16
+    renamable $x8 = LDRXui $sp, 1 :: (load (s64) from %stack.0)
+    STRXui $xzr, $sp, 1 :: (store (s64) into %stack.0)
```

Comment:
Just to confirm my understanding, this is deliberately clobbering the stack location to force LDV to move the location to the register, yes?

Reply:
Yes I am! I wanted to make sure that even when we clobber, LDV can recover the value, as it also exists in`$x8`